blob/store/structured/mongo/blob_repository.go

```diff
@@ -0,0 +1,431 @@
+package mongo
+
+import (
+	"context"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/tidepool-org/platform/blob"
+	blobStoreStructured "github.com/tidepool-org/platform/blob/store/structured"
+	"github.com/tidepool-org/platform/errors"
+	"github.com/tidepool-org/platform/log"
+	"github.com/tidepool-org/platform/page"
+	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/request"
+	storeStructuredMongo "github.com/tidepool-org/platform/store/structured/mongo"
+	structureValidator "github.com/tidepool-org/platform/structure/validator"
+	"github.com/tidepool-org/platform/user"
+)
+
+type BlobRepository struct {
+	*storeStructuredMongo.Repository
+}
+
+func (b *BlobRepository) EnsureIndexes() error {
+	return b.CreateAllIndexes(context.Background(),
+		[]mongo.IndexModel{
+			{
+				Keys: bson.D{{Key: "id", Value: 1}},
+				Options: options.Index().
+					SetUnique(true).
+					SetBackground(true),
+			},
+			{
+				Keys: bson.D{{Key: "userId", Value: 1}},
+				Options: options.Index().
+					SetBackground(true),
+			},
+			{
+				Keys: bson.D{{Key: "mediaType", Value: 1}},
+				Options: options.Index().
+					SetBackground(true),
+			},
+			{
+				Keys: bson.D{{Key: "status", Value: 1}},
+				Options: options.Index().
+					SetBackground(true),
+			},
+		})
+}
+
+func (b *BlobRepository) List(ctx context.Context, userID string, filter *blob.Filter, pagination *page.Pagination) (blob.BlobArray, error) {
+	ctx, logger := log.ContextAndLoggerWithFields(ctx, log.Fields{"userId": userID, "filter": filter, "pagination": pagination})
+
+	if ctx == nil {
+		return nil, errors.New("context is missing")
+	}
+	if userID == "" {
+		return nil, errors.New("user id is missing")
+	} else if !user.IsValidID(userID) {
+		return nil, errors.New("user id is invalid")
+	}
+	if filter == nil {
+		filter = blob.NewFilter()
+	} else if err := structureValidator.New().Validate(filter); err != nil {
+		return nil, errors.Wrap(err, "filter is invalid")
+	}
+	if pagination == nil {
+		pagination = page.NewPagination()
+	} else if err := structureValidator.New().Validate(pagination); err != nil {
+		return nil, errors.Wrap(err, "pagination is invalid")
+	}
+
+	now := time.Now()
+
+	var status []string
+	if filter.Status != nil {
+		status = *filter.Status
+	} else {
+		status = []string{blob.StatusAvailable}
+	}
+
+	result := blob.BlobArray{}
+	query := bson.M{
+		"userId": userID,
+		"status": bson.M{
+			"$in": status,
+		},
+		"deletedTime": bson.M{
+			"$exists": false,
+		},
+	}
+	if filter.MediaType != nil {
+		query["mediaType"] = bson.M{
+			"$in": *filter.MediaType,
+		}
+	}
+	opts := storeStructuredMongo.FindWithPagination(pagination).
+		SetSort(bson.M{"createdTime": -1})
```

Comment:
Should there be an index created on `createdTime`?

Reply:
unsure - this was existing code